pkg/logger/logger.go

```diff
@@ -308,6 +312,13 @@ func (l *Logger) SetReqEnabled(e bool) {
 	l.reqEnabled = e
 }
 
+// SetReverseProxy controls whether logging will trust headers that can be set by a reverse proxy.
+func (l *Logger) SetReverseProxy(e bool) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
```

Comment:
We could remove a call to defer here and explicitly unlock the mutex at the end of the function.

Reply:
I'd used a deferred unlock as that is the standard that is followed in all other `Set*` functions in the logger.

`defer` has a small performance penalty and shouldn't be used in a very hot code path but for this initialisation code I don't think it's relevant to worry about - compared to the benefit of ensuring unlock is always called even if the function logic becomes more complicated in the future.

I'm not strongly opinionated on this either way but think the package should be consistent for all methods whether explicit calls or use of deferred unlocks is decided.